pkg/tuf/client.go

```diff
@@ -0,0 +1,618 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tuf
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/theupdateframework/go-tuf/client"
+	tuf_leveldbstore "github.com/theupdateframework/go-tuf/client/leveldbstore"
+	"github.com/theupdateframework/go-tuf/data"
+	"github.com/theupdateframework/go-tuf/util"
+)
+
+const (
```

Comment:
I think these defaults need to be set in cosign, and stuff in sigstore pkg/ should be agnostic of TUF locations

I think that means that `Initialize` and `NewFromEnv` should be in cosign, because their internal callers resolve the defaults.

Reply:
How should a non-cosign client get the fulcio roots in that case? Is there some other better way to initialize a TUF client that can get passed to sigstore's `fulcioroots.Get` and `fulcioroots.GetIntermediates`, that doesn't rely on cosign-specific defaults?

(I had assumed that cosign's defaults should become sigstore's defaults, and trickle down to anything interested in getting Fulcio's roots. Cosign had extra behavior based on the presence of the `COSIGN_ROOT_FILE` env var, which was moved back into cosign for now)